scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/CertificateMessage.java

```diff
@@ -308,6 +310,17 @@ public void verifyCertificate(X509Certificate[] trustedCertificates) throws Hand
 				AlertMessage alert = new AlertMessage(AlertLevel.FATAL, AlertDescription.BAD_CERTIFICATE, getPeer());
 				throw new HandshakeException("Certificate chain could not be validated", alert);
 			}
+		} if (rawPublicKeyBytes != null) {
+		    RawPublicKeyIdentity rpk = new RawPublicKeyIdentity(getPublicKey());
+		    for (RawPublicKeyIdentity trusted : trustedRPKs) {
```

Comment:
Did you check, if "contains" works? If the RPKs are already included in a Set, this may have a chance to work.

Reply:
Yes that should work (since RawPublicKeyIdentity implments equals()).